utils/disk_space_monitor.hh

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <chrono>
+#include <filesystem>
+
+#include <boost/signals2/connection.hpp>
+#include <boost/signals2/signal_type.hpp>
+#include <boost/signals2/dummy_mutex.hpp>
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/core/timer.hh>
+#include <seastar/util/optimized_optional.hh>
+
+#include "seastarx.hh"
+#include "utils/assert.hh"
+#include "utils/log.hh"
+#include "utils/updateable_value.hh"
+
+using namespace std::chrono_literals;
+namespace bs2 = boost::signals2;
+
```

Comment:
Do not add short names in headers. They spread everywhere.

Reply:
ok

Reply:
I also advise against including namespace in header since it can cause conflict. It happened to me :)

Reply:
> Do not add short names in headers. They spread everywhere.

Dropped in v9 ([733ca78](https://github.com/scylladb/scylladb/commit/733ca78ff5d46303b1250c9c595f7668083a239d))